### ISSUE TITLE
Sound Asset loading crash

### ISSUE THREAD

**Issue by jabdoa2** [missionpinball/mpf-mc#120]
```
Traceback (most recent call last):
  File "/data/home/jan/cloud/flipper/src/mpf-mc/mpfmc/commands/mc.py", line 152, in __init__
    thread_stopper=thread_stopper).run()
  File "/usr/local/lib/python3.5/dist-packages/kivy/app.py", line 828, in run
    runTouchApp()
  File "/usr/local/lib/python3.5/dist-packages/kivy/base.py", line 487, in runTouchApp
    EventLoop.window.mainloop()
  File "/usr/local/lib/python3.5/dist-packages/kivy/core/window/window_sdl2.py", line 619, in mainloop
    self._mainloop()
  File "/usr/local/lib/python3.5/dist-packages/kivy/core/window/window_sdl2.py", line 362, in _mainloop
    EventLoop.idle()
  File "/usr/local/lib/python3.5/dist-packages/kivy/base.py", line 327, in idle
    Clock.tick()
  File "/usr/local/lib/python3.5/dist-packages/kivy/clock.py", line 515, in tick
    self._process_events()
  File "/usr/local/lib/python3.5/dist-packages/kivy/clock.py", line 647, in _process_events
    event.tick(self._last_tick, remove)
  File "/usr/local/lib/python3.5/dist-packages/kivy/clock.py", line 406, in tick
    ret = callback(self._dt)
  File "/data/home/jan/cloud/flipper/src/mpf-mc/mpfmc/uix/display.py", line 122, in _displays_initialized
    self.mc.displays_initialized()
  File "/data/home/jan/cloud/flipper/src/mpf-mc/mpfmc/core/mc.py", line 218, in displays_initialized
    self._init()
  File "/data/home/jan/cloud/flipper/src/mpf-mc/mpfmc/core/mc.py", line 261, in _init
    self.events.process_event_queue()
  File "/data/home/jan/cloud/flipper/src/mpf/mpf/core/events.py", line 502, in process_event_queue
    **event[3])
  File "/data/home/jan/cloud/flipper/src/mpf/mpf/core/events.py", line 438, in _process_event
    result = handler.callback(**merged_kwargs)
  File "/data/home/jan/cloud/flipper/src/mpf/mpf/core/assets.py", line 195, in _create_assets
    self._create_assets_from_disk(config=mode.config, mode=mode)
  File "/data/home/jan/cloud/flipper/src/mpf/mpf/core/assets.py", line 304, in _create_assets_from_disk
    config=config[ac['disk_asset_section']][asset])
  File "/data/home/jan/cloud/flipper/src/mpf-mc/mpfmc/assets/sound.py", line 95, in __init__
    track = self.machine.sound_system.audio_interface.get_track_by_name(self.config['track'])
KeyError: 'track'
```

Happens when putting a sounds folder with a sound into a mode.

**Comment by qcapen** [comment]
Part of this bug has already been fixed in the audio-new-feature-dev branch.  A better error message is generated: "Sound must have a valid track name.  Could not create sound 'name' asset."

The next part of the issue is related to how asset defaults propagate into mode folders.  I'm currently investigating...

**Comment by qcapen** [comment]
Mode sounds are working in 0.31 (I have added a mode sound test).  The above error will not happen once the new audio code is merged into dev (hope to have it to a stable point in the next few days).

**Comment by qcapen** [comment]
Fixed in 0.31
